### PR TITLE
logging view UI work and improved filtering

### PR DESCRIPTION
**Pull request by devoncarew**
- tweak the UI of the logging view to better match the look of other pages
- remove an opacity animation from the logging pages detail area - overall I don't think this animation helped the page
- add more sophisticated filtering to the page - support negative filter terms (`foo` matches all items w/ 'foo`; `-bar` matches all items that do not contain 'bar')

Screen Shot 2020-04-19 at 6 25 47 PM

### CODE REVIEW

File: packages/devtools_app/lib/src/core/filtering.dart [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        return false;
+      }
+    }
+
+    return true;
+  }
+}
+
+class FilterItem {
+  FilterItem(this.text, [this.negative = false]);
```
Comment (jacob314) [inline packages/devtools_app/lib/src/core/filtering.dart:54]: nit: would negative be clearer as a named parameter?

File: packages/devtools_app/test/core/filtering_test.dart [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      expect(applyFilter(filter, items), ['foo']);
+      expect(
+          applyFilter(filter, ['foo', 'foofoo', 'foobar']), ['foo', 'foofoo']);
+    });
+  });
+}
+
+List<String> applyFilter(Filter filter, List<String> items) {
+  return items.where((element) {
+    final m = filter.matches((String text) => element.contains(text));
```
Comment (jacob314) [inline packages/devtools_app/test/core/filtering_test.dart:64]: nit. remove local variable `m` that doesn't add value.

File: packages/devtools_app/lib/src/logging/flutter/logging_screen.dart [hunk 3]
```diff
@@ -236,35 +223,16 @@ class _LogDetailsState extends State<LogDetails>
...
-              ),
-              Opacity(
-                opacity: 1 - crossFade.value,
-                child: _buildContent(context, _oldLog),
-              ),
-            ],
-          ),
-        );
-      },
+    return Container(
```
Comment (jacob314) [inline packages/devtools_app/lib/src/logging/flutter/logging_screen.dart:233]: The crossFade was intentionally removed, right?

File: packages/devtools_app/lib/src/core/filtering.dart (hunk 1 above)
Comment (devoncarew) [reply packages/devtools_app/lib/src/core/filtering.dart]: yup - changed

File: packages/devtools_app/test/core/filtering_test.dart (hunk 2 above)
Comment (devoncarew) [reply packages/devtools_app/test/core/filtering_test.dart]: removed (I'd added this while debugging the tests)

**Pull request merged**